File: 0567-permutation-in-string/0567-permutation-in-string.py
class Solution:
    def checkInclusion(self, s1: str, s2: str) -> bool:
        l, r = 0, len(s1) - 1

        s1 = sorted(s1)

        while r < len(s2):
            temp = s2[l:r + 1]
            temp = sorted(temp)

            if s1 == temp: return True
            
            l += 1
            r += 1
        
        return False

File: 0567-permutation-in-string/0567-permutation-in-string.js
/**
 * @param {string} s1
 * @param {string} s2
 * @return {boolean}
 */
var checkInclusion = function(s1, s2) {
    let [l, r] =[0, s1.length - 1];
    
    s1 = s1.split("");
    s1.sort();
    s1 = s1.join("");
    
    while (r < s2.length){
        let temp = s2.slice(l, r + 1).split("");
        temp.sort();
        
        if (s1 === temp.join("")){
            return true;
        };
        
        l ++;
        r ++;
    }
    
    return false;
};